src/op_analytics/datapipeline/chains/mapping_utils.py

```diff
@@ -0,0 +1,372 @@
+from pathlib import Path
```

Comment:
is the right spot to handle for chain migration:
e.g., metadata only applies for a set date range.

Options:
- Start/end are null = don't care about dates
- Celo example: L1 ended on day X, L2 started on day Y
- Maybe we add options for block range and date range? (it depends how much granularity we have).

Reply:
could also be like "DA Layer migration" - I feel like migrations are the use case